### DISCUSSION TITLE
assertViewIs test fails in v3

### DISCUSSION THREAD

**Discussion by kolydart** [livewire/livewire#6840] [7. Bugs (archived)] [answered]
I have an issue after upgrading my installation to Livewire 3.
Although the browser displays the view without error, the test `$response->assertViewIs()` fails. 
It seems that the returned type from view is string instead of a view object. 

<div type='discussions-op-text'>

### Livewire version

3.0.4

### Which PHP version are you using?

PHP 8.1

### Repository URL

https://github.com/kolydart/livewire-3-assertViewIs-debug

### Steps To Reproduce

1. Download [repository](https://github.com/kolydart/livewire-3-assertViewIs-debug)
2. Set encryption kay & sql credentials in .env.testing: `cp .env.example .env && php artisan key:generate && mv .env .env.testing && nano .env.testing`
3. Install dependencies: `php composer install`
4. Run tests: `php artisan test`

The test fails as follows:

```
     FAILED  Tests\Feature\TaskControllerTest > user can index model                                         AssertionFailedError
  The response is not a view.

  at tests/Feature/TaskControllerTest.php:24
     20▕         $response = $this->get(route("tasks.index"));
     21▕         $response->assertSessionHasNoErrors();
     22▕         $response->assertSuccessful();
     23▕
  ➜  24▕         $response->assertViewIs("tasks.index");
     25▕
     26▕     }
     27▕
```

### Additional information

The test does not fail in any of the following cases:
1. if the livewire component is disabled
2. if no controller is used (see routes/web.php for using Route::view(), instead of Route::get())

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.</div>

I am not sure this is the same as [#6212]( https://github.com/livewire/livewire/discussions/6212).

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Comment by jacksontong** [comment]
I have same issue.

**Comment by MarcoCazzaro** [comment]
Same here. I also have problems with `assertHas()`.

**Comment by mabdullahsari** [comment]
We had to drop a lot of `assertViewIs` and `assertViewHas` assertions during our Livewire 3 upgrade, which is a huge disadvantage for the long run.

Please take a look at this problem asap. Thanks!

**Comment by titantwentyone** [comment]
Just came across this issue today. In my app, I am using Filament and specifically the notifications package on the front-end. The docs there state that we should insert `@livewire('notifications')` at the botom of the page. This seems to be why my `assertViewIs` tests are not working. Removing this allows the assertions to pass but obviously I need it for the app to function as needed.

EDIT: I've found that if you call the view prior to the test, the assertions pass. I'm unsure of the internals of Livewire but I beiieve there is a cache in play here which will be set if you call the view before:

```
$this->get('/');

$this->get('/')
    ->assertSuccessful()
    ->assertViewIs('my-view');
```

**Comment by da-mask** [comment]
I've got around this by continuing to manually include `@livewireScripts` & ` @livewireStyles` assets in my app's layout file. I'm not really sure how/why that works, but it you are wanting to keeps those tests, you can do this until this is worked out.

**Reply by kolydart** [reply, marked as answer]
This also worked in my case. Actually, I also had to include `@livewireScripts` in app's layout file. Tests now pass green.

I added the following:
`    @if (App::environment() == 'testing')
        @livewireScripts
    @endif
`

**Comment by rynomars** [comment]
This issue has been resolved in v3.4.0 with PR #7758. You can now use the auto injection of the livewire js asset and use assertViewIs and assertViewHas on a laravel response.
